### ISSUE TITLE
Combobox + NextJS Link doesn't work with keyboard navigation (Enter)

### ISSUE THREAD

**Issue by Strijdhagen** [tailwindlabs/headlessui#1899]
### Discussed in https://github.com/tailwindlabs/headlessui/discussions/1821

<div type='discussions-op-text'>

<sup>Originally posted by **Strijdhagen** September  5, 2022</sup>
Hi All,

I'm using the following workaround this use Combobox with Link from NextJS, which works great when using mouse navigation: https://headlessui.com/react/menu#integrating-with-next-js

However, when using keyboard navigation, "Enter" specifically, it selects the correct choice but it doesn't follow the link.
I use the TailwindUI Combobox with secondary text in this example: https://tailwindui.com/components/application-ui/forms/comboboxes#component-0834ed9ce704a78cc0045ef92f4af723.

Does anyone have a solution for this>

```

            {filteredPeople.map((person) => (
              <MyLink 
              href={"}>
                <Combobox.Option
                key={person.name}
                value={person}
                className={({ active }) =>
                  classNames(
                    'relative cursor-default select-none py-2 pl-3 pr-9',
                    active ? 'bg-slate-600 text-white' : 'text-slate-900'
                  )   
                
                }
              >
                
                {({ active, selected }) => (
                  <>
                    <div className="flex">
                      <span className={classNames('truncate', selected && 'font-semibold')}>{person.name}</span>
                      <span
                        className={classNames(
                          'ml-2 truncate text-slate-500',
                          active ? 'text-slate-200' : 'text-slate-500'
                        )}
                      >
                        {person.name}
                      </span>
                    </div>
                    
                    {selected && (
                      <span
                        className={classNames(
                          'absolute inset-y-0 right-0 flex items-center pr-4',
                          active ? 'text-white' : 'text-slate-600'
                        )}
                      >
                        <CheckIcon className="h-5 w-5" aria-hidden="true" />
                      </span>
                    )}
                  </>
                )}
              </Combobox.Option>
            </MyLink>
```</div>
